### DISCUSSION TITLE
Typescript for onConnect connectionParams?

### DISCUSSION THREAD

**Discussion by theogravity** [enisdenjo/graphql-ws#310] [Q&A] [answered]
What's the generic to use for defining the `connectionParams` on context?

```
useServer({
  onConnect: async (ctx) => {
     // ctx.connectionParams.token is of type unknown
     const token = ctx.connectionParams.token
  }
})
```

**Comment by enisdenjo** [comment, marked as answer]
There was none at the time of asking. The idea was that the server has to deal with clients that may not comply with the required `connectionParams` too, so I left the generic out.

However, you indeed expect _most_ of the clients to send over the proper params, so why not have a generic! Landed in [v5.6.0](https://github.com/enisdenjo/graphql-ws/releases/tag/v5.6.0) (through https://github.com/enisdenjo/graphql-ws/pull/311). Your example becomes:
```ts
useServer<{ token: string }>(
  {
    onConnect: async (ctx) => {
      // connectionParams can be undefined because you shouldnt
      // rely on all clients on the internet to send them
      const token: string | undefined = ctx.connectionParams?.token;
    },
  },
  ...
);
```

**Reply by theogravity** [reply]
Thank you!
